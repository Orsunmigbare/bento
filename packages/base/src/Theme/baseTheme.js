import { ThemeManager } from "@staccx/theme"

export const theme = {
  name: "Default theme",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px",
    micro: "3px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px",
    breakout: "640px"
  },
  color: {
    white: "#fff",
    bg: "#fff",
    subtleHover: "#F7F7FF",
    bgGray: "#f7f7ff",
    line: "#EBEBF2",
    gray: "#767676",
    wcag: "#777",
    disabled: "#BABABA",
    primary: "#00a190",
    secondary: "#F5A623",
    text: "#2b2b2b",
    negative: "#ea526f",
    positive: "#4A90E2",
    warning: "#c54059",
    black: "#000",
    green: "#00a190" // for toggle eks
  },
  fontFamily: {
    heading: "sans-serif",
    body: "sans-serif"
  },
  font: {
    h1: "36px",
    h2: "26px",
    h3: "22px",
    h4: "18px",
    h5: "16px",
    h6: "14px",
    base: "14px",
    tiny: "12px",
    input: "16px",
    huge: "44px"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

const manager = new ThemeManager(theme)
export default manager
