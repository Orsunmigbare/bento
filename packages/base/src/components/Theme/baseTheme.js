export const theme = {
  name: "StaccXBase",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px"
  },
  color: {
    bg: "#fff",
    bgGray: "#f7f7ff",
    primary: "#00a190",
    gray: "#767676",
    text: "#2b2b2b",
    white: "#fff",
    red: "#ea526f",
    green: "#00a190",
    blue: "#988EFB",
    yellow: "#f4da87",
    line: "#EBEBF2",
    wcag: "#777",
    disabled: "#BABABA",
    warning: "#c54059",
    black: "#000",
    subtleHover: "#F7F7FF"
  },
  font: {
    size: {
      h1: "36px",
      h2: "26px",
      h3: "22px",
      h4: "18px",
      h5: "16px",
      h6: "14px",
      base: "14px",
      tiny: "12px",
      input: "16px",
      huge: "44px"
    },
    heading: "Libre Baskerville",
    body: "Libre Franklin"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

export default theme
