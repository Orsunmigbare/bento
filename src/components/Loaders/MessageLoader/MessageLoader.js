import React, { useState, useEffect } from "react"
import PropTypes from "prop-types"
import styled, { css } from "styled-components"
import useInterval from "../../../hooks/useInterval"
import { applyVariants, color, font, mapProp, spacing } from "../../../theming"
import Loading from "../Loading/Loading"
import themeProps from "./MessageLoader.themeProps"

const MessageLoader = ({
  messages = [],
  delay = 5000,
  animation = 500,
  randomOrder = false,
  messagePosition = "bottom",
  textColor,
  variant
}) => {
  const [message, setMessage] = useState(null)
  const [lastMessage, setLastMessage] = useState(null)
  const [index, setIndex] = useState(
    randomOrder ? Math.floor(Math.random() * messages.length) : 0
  )
  useEffect(() => {
    setLastMessage(message)
    setMessage(messages[index])
  }, [messages, index])

  useInterval(() => {
    if (randomOrder) {
      setIndex(Math.floor(Math.random() * messages.length))
    } else if (index + 1 >= messages.length) {
      setIndex(0)
    } else {
      setIndex(index + 1)
    }
  }, delay)
  return (
    <Container messagePosition={messagePosition} variant={variant}>
      <Loading idleTime={0} />
      <MessagesContainer variant={variant}>
        {lastMessage && (
          <PreviousMsg
            textColor={textColor}
            key={lastMessage}
            animation={animation}
            variant={variant}
          >
            {lastMessage}
          </PreviousMsg>
        )}
        {message && (
          <CurrentMsg
            textColor={textColor}
            key={message}
            animation={animation}
            variant={variant}
          >
            {message}
          </CurrentMsg>
        )}
      </MessagesContainer>
    </Container>
  )
}

const Container = styled.div`
  display: flex;
  justify-content: space-around;
  flex-direction: ${mapProp(
    "messagePosition",
    {
      bottom: "column",
      top: "column-reverse",
      left: "row",
      right: "row-reverse"
    },
    "column"
  )};
  width: 100%;
  align-items: center;
  ${applyVariants(themeProps.innerContainer)};
`

const MessagesContainer = styled.div`
  margin: ${spacing.small} 0 ${spacing.large} 0;
  position: relative;
  width: 100%;
  ${applyVariants(themeProps.messagesContainer)};
`

const messageStyle = css`
  color: ${p => p.textColor ?? color.text};
  font-size: ${font.h2};
  text-align: center;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  padding: 0 ${spacing.medium};

  @media (max-width: 425px) {
    font-size: ${font.input};
  }

  ${applyVariants(themeProps.message)};
`

const PreviousMsg = styled.div`
  ${messageStyle};

  animation: fadeout ${p => p.animation}ms linear;
  animation-fill-mode: forwards;
  @keyframes fadeout {
    0% {
      opacity: 1;
    }
    100% {
      opacity: 0;
    }
  }
  ${applyVariants(themeProps.previousMessage)};
`

const CurrentMsg = styled.div`
  ${messageStyle};

  animation: fadein ${p => p.animation}ms linear;
  @keyframes fadein {
    0% {
      opacity: 0;
    }
    100% {
      opacity: 1;
    }
  }
  ${applyVariants(themeProps.currentMessage)};
`

MessageLoader.propTypes = {
  /**
   * Messages to be displayed
   */
  messages: PropTypes.array,
  /**
   * Where will the message be placed
   */
  messagePosition: PropTypes.oneOf(["bottom", "top", "left", "right"]),
  /**
   * How long between each message to display
   */
  delay: PropTypes.number,
  /**
   * How many ms of animation in transition between messages
   */
  animation: PropTypes.number,
  /**
   * Variant
   */
  variant: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.string,
    PropTypes.func,
    PropTypes.object
  ]),
  /**
   * Should the messages ble played randomdly?
   */
  randomOrder: PropTypes.bool
}
MessageLoader.themeProps = themeProps

export default MessageLoader
