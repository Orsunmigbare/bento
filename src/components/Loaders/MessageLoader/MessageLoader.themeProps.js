import { themePropTypes, createVariantsFunctionFactory } from "../../../theming"

const themeProps = {
  container: {
    name: "MESSAGELOADER_INNERCONTAINER",
    description: "",
    type: themePropTypes.style
  },
  messagesContainer: {
    name: "MESSAGELOADER_MESSAGESCONTAINER",
    description: "",
    type: themePropTypes.style
  },
  message: {
    name: "MESSAGELOADER_MESSAGESSTYLE",
    description: "",
    type: themePropTypes.style
  },
  previousMessage: {
    name: "MESSAGELOADER_PREVIOUSMESSAGE",
    description: "",
    type: themePropTypes.style
  },
  currentMessage: {
    name: "MESSAGELOADER_CURRENTMESSAGE",
    description: "",
    type: themePropTypes.style
  }
}

export default createVariantsFunctionFactory(themeProps)
