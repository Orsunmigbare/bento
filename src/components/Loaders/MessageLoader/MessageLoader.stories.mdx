import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import MessageLoader from "./MessageLoader"
import {messages} from "../loaders.messages"


<Meta title="Components|Loaders/MessageLoader" component={MessageLoader}/>

# MessageLoader
MessageLoader is like the FullScreenLoader only that it only takes up the full amount of its parent.

```javascript
import { MessageLoader } from "@staccx/bento"
```

## Standard MessageLoader
This is the standard MessageLoader

  <Story name="Standard">
   <MessageLoader messages={messages} random/>
  </Story>


## Props for MessageLoader
<Props of={MessageLoader} />
