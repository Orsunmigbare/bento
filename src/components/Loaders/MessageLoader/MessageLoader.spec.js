import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../../theming/themes/baseTheme"
import MessageLoader from "./MessageLoader"
import { messages } from "../loaders.messages"

describe("FullScreenLoader", () => {
  describe("Snapshots", () => {
    it("Messages", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <MessageLoader messages={["hei"]} />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("Multiple words", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <MessageLoader messages={messages} delay={1000} />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("invalid Messages", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <MessageLoader messages="hei" />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
