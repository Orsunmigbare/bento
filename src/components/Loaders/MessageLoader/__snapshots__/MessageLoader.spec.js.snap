// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageLoader Snapshots Messages 1`] = `
<div
  className="sc-AykKD ggtnrM"
>
  <div
    className="sc-AykKE fffipx"
  />
</div>
`;

exports[`MessageLoader Snapshots Multiple words 1`] = `
<div
  className="sc-AykKD ggtnrM"
>
  <div
    className="sc-AykKE fffipx"
  >
    <div
      className="sc-AykKG hyXFcy"
    >
      Reticulating splines...
    </div>
  </div>
</div>
`;

exports[`MessageLoader Snapshots invalid Messages 1`] = `
<div
  className="sc-AykKD ggtnrM"
>
  <div
    className="sc-AykKE fffipx"
  />
</div>
`;
